### PR TITLE
Support APCNet

### PR DESCRIPTION
**Pull request by Junjun2016**
Support new decode_head APCHead proposed in [APCNet CVPR'19](https://openaccess.thecvf.com/content_CVPR_2019/papers/He_Adaptive_Pyramid_Context_Network_for_Semantic_Segmentation_CVPR_2019_paper.pdf).

### CODE REVIEW

File: mmseg/models/decode_heads/apc_head.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+    """Adaptive Context Module used in APCNet.
+
+    Args:
+        pool_scale (int): Pooling scale used in Adaptive Context
+            Module to extract region fetures.
+        fusion (bool): Add one conv to fuse residual feature.
+        in_channels (int): Input channels.
+        channels (int): Channels after modules, before conv_seg.
+        conv_cfg (dict|None): Config of conv layers.
+        norm_cfg (dict|None): Config of norm layers.
```
Comment (xvjiarui) [inline mmseg/models/decode_heads/apc_head.py:21]: `dict | None`

File: mmseg/models/decode_heads/apc_head.py [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+            Module. Default: (1, 2, 3, 6).
+        fusion (bool): Add one conv to fuse residual feature.
+    """
+
+    def __init__(self, pool_scales=(1, 2, 3, 6), fusion=True, **kwargs):
+        super(APCHead, self).__init__(**kwargs)
+        assert isinstance(pool_scales, (list, tuple))
+        self.pool_scales = pool_scales
+        self.fusion = fusion
+        self.acm_modules = []
```
Comment (xvjiarui) [inline mmseg/models/decode_heads/apc_head.py:129]:
```suggestion
        acm_modules = []
```

File: mmseg/models/decode_heads/apc_head.py [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+        for pool_scale in self.pool_scales:
+            self.acm_modules.append(
```
Comment (xvjiarui) [inline mmseg/models/decode_heads/apc_head.py:131]:
```suggestion
            acm_modules.append(
```

File: mmseg/models/decode_heads/apc_head.py [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+                ACM(pool_scale,
+                    self.fusion,
+                    self.in_channels,
+                    self.channels,
+                    conv_cfg=self.conv_cfg,
+                    norm_cfg=self.norm_cfg,
+                    act_cfg=self.act_cfg))
+        self.acm_modules = nn.ModuleList(self.acm_modules)
```
Comment (xvjiarui) [inline mmseg/models/decode_heads/apc_head.py:139]:
```suggestion
        self.acm_modules = nn.ModuleList(acm_modules)
```

File: mmseg/models/decode_heads/apc_head.py [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+        # [batch_size, channels, pool_scale, pool_scale]
+        pooled_x = self.pooled_redu_conv(pooled_x)
+        batch_size = x.size(0)
+        # [batch_size, pool_scale * pool_scale, channels]
+        pooled_x = pooled_x.view(batch_size, self.channels,
+                                 -1).permute(0, 2, 1).contiguous()
+        # [batch_size, h * w, pool_scale * pool_scale]
+        affinity_matrix = self.gla(x + resize(
+            self.global_info(F.adaptive_avg_pool2d(x, 1)), size=x.shape[2:])
+                                   ).permute(0, 2, 3, 1).contiguous().view(
```
Comment (xvjiarui) [inline mmseg/models/decode_heads/apc_head.py:92]: `.contiguous().view(` -> `.reshape(`

**Comment by hellock** [comment]
Task linked: [CU-hera3z APC Net](https://app.clickup.com/t/hera3z)

**Review by xvjiarui — APPROVED** [review]

**Pull request merged**